### PR TITLE
Build artifact for UI-documentation's structure

### PR DESCRIPTION
**Pull request by nhaagen**
data.json is now data.php and is built upon composer dump-autoload.

### CODE REVIEW

**Review by Amstutz — APPROVED** [review]
Thx a lot @nhaagen ! Thanks @chfsx for the artifact builder, making this overdue step finally possible!

@chfsx do you want to look over it, or is it fine if I click merge. I need this for some refactoring I need to do for the system styles in 6.0.

File: Services/Style/System/classes/Setup/class.ilUIBuildDataStructureObjective.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	{
+		return $this->data_path;
+	}
+
+
+	public function build() : Setup\Artifact
+	{
+		$crawler = new Crawler\FactoriesCrawler();
+		$entries = $crawler->crawlFactory($this->crawler_path);
+		$entries_array = json_decode(json_encode($entries), true);
```
Comment (klees) [inline Services/Style/System/classes/Setup/class.ilUIBuildDataStructureObjective.php:26]: Why encode and then decode directly afterwards?

**Review by klees — APPROVED** [review]
:+1:

LGTM in general. Thx a lot!

**Review by chfsx — APPROVED** [review]
Nice! Thanks a lot and cool someone's using the artifact-builder :-)

**Pull request merged**